### PR TITLE
Format `PatternMatchOr`

### LINKED ISSUE #6643: Format `PatternMatchOr`
Implement formatting of `a | b` patterns

```python
match x:
    case [x] | (y):
        ...
```

### PR DESCRIPTION
**Pull request by LaBatata101**
## Summary
There was only one case where I couldn't figure it out how to format properly, which is the following
```python
match x:
    case (
        (a)
        | # trailing
        ( b )
    ):
        ...
```
That code is being formatted to this
```python
match x:
    case (
        (
            a  # trailing
        )
        | (b)
    ):
        ...
```
But it should be formatted to this
```python
match x:
    case (
        (a)  # trailing
        | (b)
    ):
        ...
```

Closes #6643

## Test Plan

`cargo test`

### CODE REVIEW

**Comment by LaBatata101** [comment]
Now clippy complains about `not_yet_implemented_custom_text` never being used.

**Comment by MichaReiser** [comment]
> Now clippy complains about `not_yet_implemented_custom_text` never being used. Should this function be removed?
> 
> And look at that amount of lines removed ☺️

Finally, our last syntax is now implemented. Yes, we can remove that function.

File: crates/ruff_python_formatter/src/lib.rs [hunk 1]
```diff
@@ -167,45 +165,6 @@ pub fn format_node<'a>(
     Ok(formatted)
 }
 
-pub(crate) struct NotYetImplementedCustomText<'a> {
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/src/lib.rs:170]: Nice!

File: crates/ruff_python_formatter/src/pattern/pattern_match_or.rs [hunk 2]
```diff
@@ -2,22 +2,46 @@ use ruff_formatter::write;
...
+                return Ok(());
+            };
+
+            first.format().fmt(f)?;
+
+            for pattern in patterns {
+                let leading_value_comments = comments.leading(pattern);
+                // Format the expressions leading comments **before** the operator
+                if leading_value_comments.is_empty() {
+                    write!(f, [in_parentheses_only_soft_line_break_or_space()])?;
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/src/pattern/pattern_match_or.rs:31]: Using the `in_parentheses_only` builder here seems correct but isn't sufficient without using `optional_parentheses` in the match case formatting. I created an issue to track that work #6933

**Review by MichaReiser — APPROVED** [review]
Neat! Thanks for working on the last pattern.

**Comment by MichaReiser** [comment]
This PR improves the similarity index for CPython from cpython     0.76076 to 0.76081

**Pull request merged**